### PR TITLE
major rewrite of lemon_markets.data.streams, workflows and basic documentation

### PR DESCRIPTION
**Pull request by leonhma**
For the automation to work you have to add the pypi and testpypi keys under 'secrets' as 'TEST_PYPI_PASSWD' and 'PYPI_PASSWD' and under settings, select gh-pages to host your github-pages site. You have to have manually commited to this branch at least once (just add a space in .nojekyll)

### CODE REVIEW

File: lemon_markets/account.py [hunk 1]
```diff
@@ -42,7 +42,7 @@ class Fields(AbstractApiObjectMixin.Fields):
         currency: str
         uuid: str
 
-    def __init__(self, uuid: str, authorization_token: Union[str, "Token"] = None):
+    def __init__(self, uuid: str, authorization_token=None):
```
Comment (mkatenhusen) [inline lemon_markets/account.py:45]: please leave the old type descriptions for clarity:)

File: lemon_markets/account.py [hunk 2]
```diff
@@ -1,4 +1,4 @@
-from typing import Union
+# pylama:ignore=E501
```
Comment (mkatenhusen) [inline lemon_markets/account.py:1]: please leave the old type descriptions for clarity:)

File: lemon_markets/common/__init__.py [hunk 3]
```diff
@@ -1 +1 @@
-#__all__ = ["objects"]
```
Comment (mkatenhusen) [inline lemon_markets/common/__init__.py:1]: can be entirely removed

File: lemon_markets/common/helpers.py [hunk 4]
```diff
@@ -23,8 +24,6 @@ class AccountError(BaseError):
 
 
 class AccountObjectMixin:
-    account: "Account"
```
Comment (mkatenhusen) [inline lemon_markets/common/helpers.py:26]: please leave the old type descriptions for clarity:)

File: lemon_markets/common/helpers.py [hunk 5]
```diff
@@ -1,5 +1,6 @@
+# pylama:ignore=E501
```
Comment (mkatenhusen) [inline lemon_markets/common/helpers.py:1]: please leave the old type descriptions for clarity:)

File: lemon_markets/account.py [hunk 6]
```diff
@@ -67,7 +67,7 @@ def retrieve(self):
         return self
 
     @staticmethod
-    def list(authorization_token: Union[str, "Token"], limit: int = None, offset: int = None) -> ListIterator:
```
Comment (mkatenhusen) [inline lemon_markets/account.py:70]: please leave the old type descriptions for clarity:)

File: lemon_markets/__init__.py [hunk 7]
```diff
@@ -1,4 +1,4 @@
 default_rest_api_url: str = "https://api.lemon.markets/rest/v1/"
 default_stream_api_url: str = "https://api.lemon.markets/streams2/v1/"
 
-#__all__ = ["common", "token"]
```
Comment (mkatenhusen) [inline lemon_markets/__init__.py:4]: can be entirely removed

File: lemon_markets/common/objects.py [hunk 8]
```diff
@@ -1,3 +1,5 @@
+# pylama:ignore=E501
```
Comment (mkatenhusen) [inline lemon_markets/common/objects.py:1]: please leave the old type descriptions for clarity:)

File: lemon_markets/common/requests.py [hunk 9]
```diff
@@ -48,12 +50,12 @@ class ApiRequest:
     method: str = "GET"
     body: dict
     authorization_token: str
-    _account: "Account" = None
```
Comment (mkatenhusen) [inline lemon_markets/common/requests.py:51]: please leave the old type descriptions for clarity:)

File: lemon_markets/common/requests.py [hunk 10]
```diff
@@ -48,12 +50,12 @@ class ApiRequest:
...
+    _account = None
     _kwargs: dict
     _response: ApiResponse
 
     def __init__(self, endpoint: str, method: str = "GET", body: dict = None,
-                 authorization_token: Union[str, "Token"] = None, url_params: dict = {}, **kwargs):
```
Comment (mkatenhusen) [inline lemon_markets/common/requests.py:56]: please leave the old type descriptions for clarity:)

File: lemon_markets/common/requests.py [hunk 11]
```diff
@@ -1,3 +1,5 @@
+# pylama:ignore=E501
```
Comment (mkatenhusen) [inline lemon_markets/common/requests.py:1]: please leave the old type descriptions for clarity:)

File: lemon_markets/data/ohlc.py [hunk 12]
```diff
@@ -1,3 +1,5 @@
+# pylama:ignore=E501
```
Comment (mkatenhusen) [inline lemon_markets/data/ohlc.py:1]: please leave the old type descriptions for clarity:)

File: lemon_markets/data/ohlc.py [hunk 13]
```diff
@@ -23,10 +25,10 @@ class Fields(AbstractApiObject.Fields):
 class AbstractDataMixin(ListMixin):
 
     @classmethod
-    def _build_endpoint(cls, instrument: Union[str, "Instrument"]) -> str:
```
Comment (mkatenhusen) [inline lemon_markets/data/ohlc.py:26]: please leave the old type descriptions for clarity:)

File: lemon_markets/data/ohlc.py [hunk 14]
```diff
@@ -23,10 +25,10 @@ class Fields(AbstractApiObject.Fields):
...
+    def _build_endpoint(cls, instrument) -> str:
         raise NotImplementedError()
 
-    def latest(self, instrument: Union[str, "Instrument"], authorization_token: Union[str, "Token"] = None):
```
Comment (mkatenhusen) [inline lemon_markets/data/ohlc.py:29]: please leave the old type descriptions for clarity:)

File: lemon_markets/data/ohlc.py [hunk 15]
```diff
@@ -36,19 +38,19 @@ def latest(self, instrument: Union[str, "Instrument"], authorization_token: Unio
         self.set_data(request.response)
         return self
 
-    def retrieve(self, instrument: Union[str, "Instrument"], authorization_token: Union[str, "Token"] = None):
```
Comment (mkatenhusen) [inline lemon_markets/data/ohlc.py:39]: please leave the old type descriptions for clarity:)

File: lemon_markets/data/ohlc.py [hunk 16]
```diff
@@ -36,19 +38,19 @@ def latest(self, instrument: Union[str, "Instrument"], authorization_token: Unio
...
+    def retrieve(self, instrument, authorization_token=None):
         return self.latest(instrument=instrument, authorization_token=authorization_token)
 
     @classmethod
     def list(
              cls,
-             instrument: Union[str, "Instrument"],
```
Comment (mkatenhusen) [inline lemon_markets/data/ohlc.py:45]: please leave the old type descriptions for clarity:)

File: lemon_markets/data/ohlc.py [hunk 17]
```diff
@@ -36,19 +38,19 @@ def latest(self, instrument: Union[str, "Instrument"], authorization_token: Unio
...
+             instrument,
              ordering: str = "-date",
              date_from: Union[str, datetime.datetime] = None,
              date_until: Union[str, datetime.datetime] = None,
              limit: int = None,
              offset: int = None,
-             authorization_token: Union[str, "Token"] = None,
```
Comment (mkatenhusen) [inline lemon_markets/data/ohlc.py:51]: please leave the old type descriptions for clarity:)

File: lemon_markets/data/ohlc.py [hunk 18]
```diff
@@ -63,15 +65,15 @@ def list(
 class M1(OHLCObject, AbstractDataMixin):
 
     @classmethod
-    def _build_endpoint(cls, instrument: Union[str, "Instrument"]) -> str:
```
Comment (mkatenhusen) [inline lemon_markets/data/ohlc.py:66]: please leave the old type descriptions for clarity:)

File: lemon_markets/data/ohlc.py [hunk 19]
```diff
@@ -63,15 +65,15 @@ def list(
...
+    def _build_endpoint(cls, instrument) -> str:
         endpoint = "data/instruments/{}/candle/m1/".format(str(instrument))
         return endpoint
 
 
 class Trades(TradeObject, AbstractDataMixin):
 
     @classmethod
-    def _build_endpoint(cls, instrument: Union[str, "Instrument"] = None) -> str:
```
Comment (mkatenhusen) [inline lemon_markets/data/ohlc.py:74]: please leave the old type descriptions for clarity:)

File: lemon_markets/data/streams.py [hunk 20]
```diff
@@ -32,142 +33,186 @@ def to_representation(self):
...
+
+    Note:
+        The descriptions may be incomplete
+    '''
+    def __init__(self, message, subscribed):
         super().__init__(message)
         self.isin = self.json_content.get("isin")
         self.bid_price = self.json_content.get("bid_price")
         self.ask_price = self.json_content.get("ask_price")
-        self.date = datetime.datetime.fromtimestamp(float(self.json_content.get("date")))
```
Comment (mkatenhusen) [inline lemon_markets/data/streams.py:52]: please stick to the "date" name as it as part of our API

File: lemon_markets/data/streams.py [hunk 21]
```diff
@@ -1,16 +1,17 @@
...
 
-import websocket
+from websocket import create_connection
 
 from lemon_markets.common.errors import StreamError
 from lemon_markets.settings import DEFAULT_STREAM_API_URL
 
 
 class BaseSerializer:
-    json_content: dict = None
```
Comment (mkatenhusen) [inline lemon_markets/data/streams.py:12]: rather keep this as class attribute

File: lemon_markets/data/streams.py [hunk 22]
```diff
@@ -32,142 +33,186 @@ def to_representation(self):
...
+
+    Note:
+        The descriptions may be incomplete
+    '''
+    def __init__(self, message, subscribed):
         super().__init__(message)
         self.isin = self.json_content.get("isin")
         self.price = self.json_content.get("price")
         self.quantity = self.json_content.get("quantity")
-        self.date = datetime.datetime.fromtimestamp(float(self.json_content.get("date")))
```
Comment (mkatenhusen) [inline lemon_markets/data/streams.py:69]: keep date

File: lemon_markets/instrument.py [hunk 23]
```diff
@@ -14,8 +16,8 @@ class Fields(AbstractApiObjectMixin.Fields):
         wkn: str
 
     def __init__(self, isin: str = None,
-                 account: "Account" = None,
-                 authorization_token: Union[str, "Token"] = None,
```
Comment (mkatenhusen) [inline lemon_markets/instrument.py:18]: please leave the old type descriptions for clarity:)

File: lemon_markets/instrument.py [hunk 24]
```diff
@@ -31,7 +33,7 @@ def to_representation(self) -> str:
     def list(
             search: str = "",
             type: Union[str, list] = "",
-            authorization_token: Union[str, "Token"] = None,
```
Comment (mkatenhusen) [inline lemon_markets/instrument.py:34]: please leave the old type descriptions for clarity:)

File: lemon_markets/order.py [hunk 25]
```diff
@@ -82,11 +84,11 @@ def create(self):
     def destroy(self, raise_exception: bool = False) -> bool:
         try:
             self.check_instance()
-            request = ApiRequest(
-                endpoint="orders/{}/".format(self.uuid),
-                account=self.account,
-                method="DELETE"
-            )
```
Comment (mkatenhusen) [inline lemon_markets/order.py:89]: please leave this, deleting an order is important.

File: lemon_markets/portfolio.py [hunk 26]
```diff
@@ -1,3 +1,5 @@
+# pylama:ignore=E501
```
Comment (mkatenhusen) [inline lemon_markets/portfolio.py:1]: please remove this auto-comment in general

File: lemon_markets/strategy.py [hunk 27]
```diff
@@ -1,3 +1,5 @@
+# pylama:ignore=E501
+
```
Comment (mkatenhusen) [inline lemon_markets/strategy.py:2]: please remove this auto-comment in general

File: lemon_markets/token.py [hunk 28]
```diff
@@ -1,3 +1,5 @@
+# pylama:ignore=E501
+
```
Comment (mkatenhusen) [inline lemon_markets/token.py:2]: please remove this auto-comment in general

File: lemon_markets/transaction.py [hunk 29]
```diff
@@ -1,3 +1,5 @@
+# pylama:ignore=E501
+
```
Comment (mkatenhusen) [inline lemon_markets/transaction.py:2]: please remove this auto-comment in general

File: setup.py [hunk 30]
```diff
@@ -4,14 +4,14 @@
...
 setuptools.setup(
-    name="lemon-markets-sdk",
+    name="lemon_markets",
     version="0.0.1",
     author="Marcel Katenhusen",
     author_email="marcel@lemon.markets",
     description="Official SDK for the lemon.markets trading & market data API",
     long_description=long_description,
     long_description_content_type="text/markdown",
-    url="https://github.com/lemon-markets/lemon-markets-python-sdk",
```
Comment (mkatenhusen) [inline setup.py:14]: keep the old name

Comment (leonhma) [reply setup.py:14]: I was just thinking you want to have a python and js and whatever language sdk and lemon-markets/lemon-markets-python-sdk is just repetitive, but I'll change it...

File: lemon_markets/common/helpers.py (hunk 5 above)
Comment (leonhma) [reply lemon_markets/common/helpers.py:1]: abc.ABC is unused

File: lemon_markets/order.py (hunk 25 above)
Comment (leonhma) [reply lemon_markets/order.py:89]: okay, just dont need to assign it to variable here

File: setup.py (hunk 30 above)
Comment (leonhma) [reply setup.py:14]: dashes in a package name are not very well accepted by various linters and also cause trouble with the github workflow. ill leave the new package name ('lemon_markets') and change the repo name back to normal, ok?

**Review by leonhma — COMMENTED** [review]
pr still alive btw

**Pull request closed without merging**